src/adagrams.js

```diff
@@ -1,15 +1,152 @@
+const letterObject = {
+    'A': [9,1],
+    'B': [2,3],
+    'C': [2,3],
+    'D': [4,2],
+    'E': [12,1],
+    'F': [2,4],
+    'G': [3,2],
+    'H': [2,4],
+    'I': [9,1],
+    'J': [1,8],
+    'K': [1,5],
+    'L': [4,1],
+    'M': [2,3],
+    'N': [6,1],
+    'O': [8,1],
+    'P': [2,3],
+    'Q': [1,10],
+    'R': [6,1],
+    'S': [4,1],
+    'T': [6,1],
+    'U': [4,1],
+    'V': [2,4],
+    'W': [2,4],
+    'X': [1,8],
+    'Y': [2,4],
+    'Z': [1,10]
+}
+
+const letterPool = []
+
+// Helper Function to Build Letter Pool
+const buildPool = () => {
+
+  for (let letter in letterObject) {
+      for (let i=0; i < letterObject[letter][0]; i++) {
+          letterPool.push(letter);
+      }
+  }
+
+  return letterPool;
+
+};
+
+buildPool();
+
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+
+  const shuffledList = letterPool.sort(() => 0.5 - Math.random());
+  return shuffledList.slice(0,10);
+
+};
+
+
+// Helper function to count num times a letter appears in string/array
+const countNum = (item) => {
+  
+  const numDict = {};
+  
+  for (let letter of item) {
+
+    letter = letter.toUpperCase()
+
+    if (numDict[letter]) {
+      numDict[letter] = numDict[letter] + 1;
+    } else {
+        numDict[letter] = 1;
+    }
+  }
+
+  return numDict;
+
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  
+  const inputUpper = input.toUpperCase();
+  const numTimesInInput = countNum(input);
+  const numTimesInHand = countNum(lettersInHand);
+
+  for (let letter of inputUpper) {
+    if (lettersInHand.includes(letter)==false || (numTimesInInput[letter]!=numTimesInHand[letter])) {
+      return false;
+    } 
+  }
+
+  return true;
```

Comment:
Using your helper functions makes this problem a lot easier! Aren't frequency map so powerful?!